infra/license/license_self_verify.ts

```diff
@@ -0,0 +1,181 @@
+import {readFileSync, writeFileSync} from 'fs';
+import {EOL} from 'os';
+
+import * as licenseBlockList from "./license-blocklist.json";
+import * as packageLicenseReg from "./package-blocklist.json";
+
+
+/**
+ * Argument List
+ *  - Input path for list of used licenses
+ *  - Tag name for the verification
+ */
+var args = process.argv.slice(2)
+const inputPathOfLicenseList = args[0];
+const verificationTag = args[1];
+
+/**
+ * - WarningCount : The total number of warnings
+ * 
+ * - WanrnedLicenseUsed
+ *   - Licenses which are classified as WARN in license-blocklist.json
+ *   - These licenses should be verified manually according to each criteria
+ *     and the package should be added to package-blocklist.json
+ * - NeverChecked
+ *   - Licenses which are not found in license-blocklist.json
+ *   - These licenses should be added license-blocklist.json
+ * - UnknownLicense
+ *   - Licenses of package are not found.
+ *   - These packages should be manually checked and added to package-blocklist.json
+ */
+var warningList = {
+    "WarningCount" : 0,
+
+    "WarnedLicenseUsed" : [] as string[],
+    "NeverChecked" : [] as string[],
+    "Unknown" : [] as string[]
+};
+
+/**
+ * - CriticalCount : The total number of criticals
+ * 
+ * - ONEForbidden
+ *   - License which are forbidden in ONE
+ * - DeniedLicenseUsed
+ *   - License which are blocked by license-blocklist.json
+ */
+var criticalList = {
+    "CriticalCount" : 0,
+
+    "ONEForbidden" : [] as string[],
+    "DeniedLicenseUsed" : [] as string[]
+};
+
+/**
+ * Verification is done by following prioritized steps
+ * 
+ * 1. If package information is found in package-blocklist.json, follow the rule.
+ *    If not found, go to step 2.
+ * 2. If license is found in license-blocklist.json, follow the rule.
+ *    If not found, go to step 3.
+ * 3. If license is UNKNOWN, register the package information to package-blocklist.json.
+ *    And then go to step 1.
+ * 4. Register license information to license-blocklist.json.
+ *    And then go to step 2.
```

Comment:
> The word "block" in each json files confuses me because it seems like the files contain only the list of should-be-blocked files. How about use license.json, packages.json or something?

I agree with you :)
If you don't mind, can we discuss this suggestion at https://github.com/Samsung/ONE-vscode/pull/292/files#r817529291? :D

---

> It would be better for each json files to have README at the start of them. Or literally README.md will do. Because it's a bit hard for me to know the meaning of each attributes, e.g. what is ONE_permitted?

I think it is really good idea!
I will create `infra/license/README.md` and then write some details at that file :)
With the readme file, those long comments can be removed :D

---

> Above comment "follow the rule". What rule should I followed?

"rule" in the comment means "follow as written in the json file" but I think it is confused :(
I will apply this opinion to the README.md file which you suggested in second question!